modules/core/api/farm_api.module

```diff
@@ -53,3 +53,30 @@ function farm_api_entity_base_field_info(EntityTypeInterface $entity_type) {
 
   return $fields;
 }
+
+/**
+ * Implements hook_farm_api_allow_resource_types().
+ */
+function farm_api_farm_api_allow_resource_types() {
+
+  // Allow farmOS core entity types on behalf of the modules that provide them.
+  return [
+    'asset',
+    'asset_type',
+    'data_stream',
+    'data_stream_type',
+    'file',
+    'log',
+    'log_type',
+    'plan',
+    'plan_type',
+    'plan_record',
+    'plan_record_type',
+    'quantity',
+    'quantity_type',
+    'taxonomy_term',
+    'taxonomy_vocabulary',
+    'user',
+    'user_role',
+  ];
```

Comment:
I'm struggling a bit with the rational for some modules declaring their own allowed resource types and others being declared here in the `farm_api` module.

I thought at first it was something about modules which are commonly disabled (declare theirs separately), but then `plan`/`data_stream`/etc would be out in their respective modules...

Reply:
I think that makes sense.

To paraphrase; the things listed here are entity types which farmOS is "built of" - but which aren't themselves supposed to be "aware" that they're part of farmOS. i.e. They shouldn't have farmOS-specific hooks in them.

That brings up two points that probably aren't strictly part of this PR:

* I think that relationship (the directionality of the module dependencies) could be enforced by a test
* Perhaps that design/decision should be documented somewhere that folks digging into the codebase will tend to find/read?

Reply:
Yes that sums up the design. And yea agreed, we could document that somewhere to make it more clear. Stepping back, it might be nice to provide some kind of "deep dive" overview of the codebase in a form of documentation somewhere. We don't really provide something like that currently.

> I think that relationship (the directionality of the module dependencies) could be enforced by a test

Interesting idea. Certainly worth exploring! Although I wonder how that would be implemented. Either way I agree it is a separate issue from this PR.